protos/service_mesh.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package smp;
+
+option csharp_namespace = "Smp";
+option go_package = "spec;smp";
+option java_multiple_files = true;
+option java_outer_classname = "ServiceMeshProto";
+option java_package = "com.smp";
+option objc_class_prefix = "smp";
+option php_namespace = "Smp";
+option ruby_package = "Smp";
+
+message ServiceMeshes {
```

Comment:
Here, ServiceMeshes will represent only one mesh.
I think it should be ServiceMesh.

Reply:
This PR is for organising spec, using style-guide for protos, adding lint checks and structuring the auto-generated stuff.
I don't prefer to handle amending fields and organisation of attributes in this PR as the PR is already huge, such enhancements and chore tasks I suggest should be handled in seperate.
Given your experience with gRPC, I guess it will be best if you yourself give it a shot, will you please prefer to go over the spec and see such fallacy and fix them? 🙂

Reply:
It's in the style guide of protos to name Fields and messages that make sense. For eg if node is repeated, an array, it should be named nodes. But yeah we can fix this in future PRs.